### PR TITLE
Building an API pt. 7 - lesson 4: refactoring for future work

### PR DESCRIPTION
**Pull request by kelsey-steven-ada**
[Pt. 7 Asana ticket](https://app.asana.com/0/1205655776549324/1206697443678573/f)
[Pt. 7 Lesson 4 ticket](https://app.asana.com/0/1205655776549324/1207017851991909/f)

Adds in refactoring the blueprint name from `books_bp` to `bp`.

I plan to rename the file to better reflect the lesson contents, I will do that as a follow up PR so folks can review the content changes side-by-side here.

### CODE REVIEW

File: api-7-refactoring/refactor-validate-book.md [hunk 1]
```diff
@@ -46,68 +35,54 @@ The `Book` class should have the following functions:
...
-- usage in `update_book`
-- usage in `delete_book`
+#### Check for Tests
 
-We can see our dependencies are the functions `read_one_book`, `update_book`, and `delete_book`, so let's see what our tests look like for those functions.
-
-### Check for Tests
-
-We previously wrote tests for `read_one_book`, so we can feel good about that function's coverage, but we don't have any tests for the functions `update_book` or `delete_book`. Furthermore, we don't have tests for `validate_book` to confirm its behavior as we make modifications. Let's pause and think through nominal and edge cases for the functions `update_book`, `delete_book`, and `validate_book`. We want to list out what test cases are missing that would give us security while we refactor.
+We previously wrote tests for the routes `create_book`, `get_all_books`, and `get_one_book`, so we can feel good about their test coverage. We are missing tests for the routes `update_book` and `delete_book` though, so let's pause and think through nominal and edge cases for `update_book` `delete_book`. We want to list out what test cases are missing that would give us security while we refactor.
```
[the remark this answers is not in the document]
Comment (kelsey-steven-ada) [reply api-7-refactoring/refactor-validate-book.md]: I like that clarity of purpose for the tests =]

File: api-7-refactoring/refactor-validate-book.md [hunk 2]
```diff
@@ -262,13 +379,16 @@ def test_validate_book_invalid_id(two_saved_books):
...
+In our previous lessons' refactors we created brand new functions and were able to write out our test cases ahead of our implementation code. In this case, we're making several implementation changes to an existing function and renaming it. There are many ways we could approach testing at this point, but we want to take a path that lets us get feedback as we make changes to our existing function rather than when we think we're done. Our recommendation is to start a cycle of making a small change to the `validate_book` tests so that they fail, then updating the code until the tests pass again, until all our changes are complete and all our tests are passing.
```
[the remark this answers is not in the document]
Comment (kelsey-steven-ada) [reply api-7-refactoring/refactor-validate-book.md]: I love this for breaking up the long sentences in this paragraph.

**Review by yangashley — APPROVED** [review]
